features/features.go

```diff
@@ -261,3 +265,21 @@ func processNegativeFlags(ftrs *Features) {
 		ftrs.UseRegistryForWithDocker = false
 	}
 }
+
+// WithContext adds the current *Features into the given context and returns a new context.
+// Trying to add the *Features to the context more than once will result in an error.
+func (f *Features) WithContext(ctx context.Context) (context.Context, error) {
+	if _, ok := ctx.Value(ctxKey{}).(*Features); ok {
```

Comment:
we *should* just be able to do `if ctx.Value(ctxKey{}) != nil {`. since no other package can use the `ctxKey` type, no other package could assign a value - so doing a type assertion isn't necessary.